### DISCUSSION TITLE
Flutter Android build: Automate the upload of debug symbols zip

### DISCUSSION THREAD

**Discussion by orkun1675** [codemagic-ci-cd/codemagic-docs#2900] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

-

### Steps to reproduce

This was previously brought up in https://github.com/orgs/codemagic-ci-cd/discussions/1729.

I have written the following script to automate the generation of the debug symbols archive:

```bash
rm -rf build/app/intermediates/stripped_native_libs/release/out/lib/.crashlytics \
&& cd build/app/intermediates/stripped_native_libs/release/out/lib/ && zip -r ../../../../../outputs/bundle/release/symbols.zip . && cd - \
&& mkdir -p build/$1/android_build/ \
&& cp build/app/outputs/bundle/release/symbols.zip build/$1/android_build/ \
&& echo "Upload the following debug symbols to Play Console: 'build/$1/android_build/symbols.zip'" \
|| echo "Build for Android failed!"
```

Is there a way for us to pass this file to Codemagic to upload it to Google Play?

If there is already an issue/bug tracking this feature request PLMK.

Thank you!

### Expected results

N/A

### Actual results

N/A

**Comment by dam-ease** [comment]
Hello @orkun1675,
Usually, the debug symbols should be automatically included when publishing your `.aab` file to the Play Store. This is what is currently supported in Codemagic; hence, there is no way to pass this file to Codemagic for uploading it to Google Play. However, I see in Flutter that this seems to be a known [issue](https://github.com/flutter/flutter/issues/98773), and it looks like the problem occurs after Android Gradle Plugin version 4.1.3. There appears to be no final solution regarding this issue, so you have to do the uploads manually pending a fix.

**Comment by orkun1675** [comment]
Thanks for the response Damilola.

Per [Maven](https://mvnrepository.com/artifact/com.android.tools.build/gradle) AGP `4.1.3` was released back in March 2021. I don't think Flutter will be resolving this issue anytime soon.

I saw that there is a Google Play Developer API endpoint that allows uploading Native Debug Symbols: https://developers.google.com/android-publisher/api-ref/rest/v3/edits.deobfuscationfiles/upload with `deobfuscationFileType=nativeCode `.

Would it be possible to file a feature request to support this?

Generating the `symbols.zip` file is quite straightforward (see my script above). I imagine it would save a lot of time for many Flutter devs to support this.

Do people really care?
* https://github.com/flutter/flutter/issues/98773 has 21 upvotes.
* https://stackoverflow.com/questions/62568757/playstore-error-app-bundle-contains-native-code-and-youve-not-uploaded-debug/79104473#79104473 has hundreds of engagements
* https://www.youtube.com/watch?v=3ieKwhmQg7I&t=405s&ab_channel=iRekhaTechSolutions has thousands of views

**Reply by dam-ease** [reply]
Hello @orkun1675,
Thanks for sharing these details!
I think this could pass as a possible feature request, it will be great if you could share all the details here - https://github.com/orgs/codemagic-ci-cd/discussions/new?category=ideas

**Reply by orkun1675** [reply]
Thank you. I've filed https://github.com/orgs/codemagic-ci-cd/discussions/2921 and will resolve this discussion.

**Comment by Gaurav-CareMonitor** [comment]
@dam-ease Any updates on this?
